File: lib/ts/src/internal/types/VFieldDeclaration.ts
import { VTypeDeclaration } from './VTypeDeclaration';

export class VFieldDeclaration {
    fieldName: string;
    typeDeclaration: VTypeDeclaration;
    optional: boolean;

    constructor(fieldName: string, typeDeclaration: VTypeDeclaration, optional: boolean) {
        this.fieldName = fieldName;
        this.typeDeclaration = typeDeclaration;
        this.optional = optional;
    }
}

File: lib/py/telepact/internal/types/VFieldDeclaration.py
from typing import Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from .VTypeDeclaration import VTypeDeclaration


class VFieldDeclaration:
    def __init__(
            self,
            field_name: str,
            type_declaration: 'VTypeDeclaration',
            optional: bool
    ) -> None:
        self.field_name = field_name
        self.type_declaration = type_declaration
        self.optional = optional
